### PR TITLE
[v2] Student Blog link route

### PR DESCRIPTION
**Pull request by YoganshSharma**
- Added `/student/bloglink/` (POST) endpoint. [Issue #126]
- Added tests for `/student/bloglink/`

### CODE REVIEW

File: controllers/student_test.go [hunk 1]
```diff
@@ -122,3 +122,85 @@ func TestStudentRegOK(t *testing.T) {
...
+	reqFields := controllers.StudentBlogLinkFields{Username: testUsername, BlogLink: "https://grugbrain.dev/"}
+	req = createStudentBlogLinkRequest(&reqFields)
+	req.Header.Add("Bearer", testJwt)
+
+	res := executeRequest(req, db)
+
+	expectStatusCodeToBe(t, res, http.StatusOK)
+}
+
+// Test an existing user registration request to /student/form/ with proper authentication and input
```
Comment (harshkhandeparkar) [inline controllers/student_test.go:162]: `/student/bloglink/` :)

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: controllers/student_test.go [hunk 2]
```diff
@@ -122,3 +122,115 @@ func TestStudentRegOK(t *testing.T) {
...
+	_ = executeRequest(req, db)
+
+	// Execute the bloglink request
+	reqFields := controllers.StudentBlogLinkReqFields{Username: testUsername, BlogLink: "https://grugbrain.dev/"}
+	req = createStudentBlogLinkRequest(&reqFields)
+	req.Header.Add("Bearer", testJwt)
+
+	res := executeRequest(req, db)
+
+	expectStatusCodeToBe(t, res, http.StatusOK)
```
Comment (harshkhandeparkar) [inline controllers/student_test.go:189]: Can you add a response body check too?

File: controllers/student_test.go [hunk 3]
```diff
@@ -122,3 +122,115 @@ func TestStudentRegOK(t *testing.T) {
...
+
+// Test request to /student/bloglink/ with invalid jwt
+func TestStudentBloglinkInvalidAuth(t *testing.T) {
+	testRequestInvalidAuth(t, "POST", "/student/bloglink/")
+}
+
+// Test request to /student/bloglink/ with session hijacking attempt
+func TestStudentBloglinkSessionHijacking(t *testing.T) {
+	// Generate a jwt secret key for testing
+	setTestJwtSecretKey()
```
Comment (harshkhandeparkar) [inline controllers/student_test.go:151]: Please defer unset this key.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]
Some minor changes

File: controllers/student_test.go (hunk 2 above)
Comment (YoganshSharma) [reply controllers/student_test.go:190]: done

File: controllers/student_test.go (hunk 3 above)
Comment (YoganshSharma) [reply controllers/student_test.go:151]: done

**Review by harshkhandeparkar — APPROVED** [review]

**Review by rajivharlalka — APPROVED** [review]
LGTM

**Pull request merged**